### PR TITLE
Factor out shared code from std.digest.toHexString overloads

### PR DESCRIPTION
**Pull request by benjones**
Pull out the duplicated code from the static and dynamic array overloads

(first PR from me.  LMK if this type of refactoring is useful or if there are other modules worth digging into)

### CODE REVIEW

File: std/digest/package.d [hunk 1]
```diff
@@ -801,6 +747,7 @@ string toHexString(LetterCase letterCase, Order order = Order.increasing)(in uby
     return toHexString!(order, letterCase)(digest);
 }
 
+
```
Comment (wilzbach) [inline std/digest/package.d]: unrelated change

File: std/net/curl.d [hunk 2]
```diff
@@ -203,9 +203,7 @@ version (unittest)
             }
             catch (Throwable e)
             {
-                import core.stdc.stdlib : exit, EXIT_FAILURE;
-                stderr.writeln(e);
-                exit(EXIT_FAILURE); // Bugzilla 7018
+                stderr.writeln(e);  // Bugzilla 7018
```
Comment (wilzbach) [inline std/net/curl.d]: unrelated?

Comment (benjones) [reply std/net/curl.d]: Yes, looks like it's a leftover or something from the previous master commit.  What's the easiest way for me to fix these?

Comment (carun) [reply std/net/curl.d]: @benjones You can amend the commit and force push it.

**Comment by wilzbach** [comment]
@benjones you still have the commit from Ian in your list. Here's what you can do:

```
git reset --soft HEAD~2
git commit
```

This will create a new fresh commit instead of the two existing ones.
